src/realizations/catchment/Bmi_Fortran_Formulation.cpp

```diff
@@ -94,77 +90,4 @@ double Bmi_Fortran_Formulation::get_var_value_as_double(const int &index, const
- * numeric types being cast if necessary.
- *
- * The BMI spec requires for variable values to be passed to/from models via as arrays.  This function essentially
- * treats the variable array reference as if it were just a raw pointer and returns the `0`-th array value.
- *
- * @param t_index The index of the time step for which to run model calculations.
- * @param d_delta_s The duration, in seconds, of the time step for which to run model calculations.
- * @return The total discharge of the model for the given time step.
- */
-double Bmi_Fortran_Formulation::get_response(time_step_t t_index, time_step_t t_delta) {
-    if (get_bmi_model() == nullptr) {
-        throw std::runtime_error("Trying to process response of improperly created BMI C formulation.");
-    }
-    if (t_index < 0) {
-        throw std::invalid_argument("Getting response of negative time step in BMI C formulation is not allowed.");
-    }
-    // Use (next_time_step_index - 1) so that second call with current time step index still works
-    if (t_index < (next_time_step_index - 1)) {
-        // TODO: consider whether we should (optionally) store and return historic values
-        throw std::invalid_argument("Getting response of previous time step in BMI C formulation is not allowed.");
-    }
-
-    // The time step delta size, expressed in the units internally used by the model
-    double t_delta_model_units;
-    if (next_time_step_index <= t_index) {
-        t_delta_model_units = get_bmi_model()->convert_seconds_to_model_time((double)t_delta);
-        double model_time = get_bmi_model()->GetCurrentTime();
-        // Also, before running, make sure this doesn't cause a problem with model end_time
-        if (!get_allow_model_exceed_end_time()) {
-            int total_time_steps_to_process = abs((int)t_index - next_time_step_index) + 1;
-            if (get_bmi_model()->GetEndTime() < (model_time + (t_delta_model_units * total_time_steps_to_process))) {
-                throw std::invalid_argument("Cannot process BMI C formulation to get response of future time step "
-                                            "that exceeds model end time.");
-            }
-        }
-    }
-
-    while (next_time_step_index <= t_index) {
-        double model_initial_time = get_bmi_model()->GetCurrentTime();
-        set_model_inputs_prior_to_update(model_initial_time, t_delta);
-        if (t_delta_model_units == get_bmi_model()->GetTimeStep())
-            get_bmi_model()->Update();
-        else
-            get_bmi_model()->UpdateUntil(model_initial_time + t_delta_model_units);
-        // TODO: again, consider whether we should store any historic response, ts_delta, or other var values
-        next_time_step_index++;
-    }
-    return get_var_value_as_double( get_bmi_main_output_var());
-}
-
-#endif // NGEN_BMI_FORTRAN_ACTIVE
+#endif // NGEN_WITH_BMI_FORTRAN
```

Comment:
Oops, looks like a rebase mistake led me to mis-match the `#if`. I'll clear that up before this gets merged

Reply:
I think I'll let #814 deal with that, and leave it alone here to avoid merge conflicts.